Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -32,27 +32,62 @@
     @Override
     public ConceptsMergeBean prepareMergeConcepts(List<ConceptEntry> conceptEntries,
             ConceptsMergeBean conceptsMergeBean) {
+        boolean typeWarning = false;
+        boolean posWarning = false;
         for (ConceptEntry entry : conceptEntries) {
             if (entry.getWord() != null) {
-                conceptsMergeBean.getWords().add(entry.getWord());
+                conceptsMergeBean.setWord(conceptsMergeBean.getWord() == null
+                        || conceptsMergeBean.getWord().length() < entry.getWord().length() ? entry.getWord()
+                                : conceptsMergeBean.getWord());
+                
             }
-            if (entry.getDescription() != null) {
-                conceptsMergeBean.getDescriptions().add(entry.getDescription());
+
+            if (conceptsMergeBean.getSelectedPosValue() == null) {
+                conceptsMergeBean.setSelectedPosValue(entry.getPos().toLowerCase());
+            } else if (!conceptsMergeBean.getSelectedPosValue().equalsIgnoreCase(entry.getPos())) {
+                posWarning = true;
+            }
+
+            if (entry.getDescription() != null && !entry.getDescription().isEmpty()) {
+                StringBuilder builder = new StringBuilder(
+                        conceptsMergeBean.getDescriptions() != null ? conceptsMergeBean.getDescriptions() : "");
+                builder.append("<p>");
+                builder.append(entry.getDescription().trim());
+                builder.append("</p>");
+                conceptsMergeBean.setDescriptions(builder.toString());
             }
+
             if (entry.getSynonymIds() != null) {
                 conceptsMergeBean.getSynonymsids().addAll(Arrays.asList(entry.getSynonymIds().split(",")));
             }
+
             if (entry.getEqualTo() != null) {
                 conceptsMergeBean.getEqualsValues().add(entry.getEqualTo());
             }
             if (entry.getSimilarTo() != null) {
                 conceptsMergeBean.getSimilarValues().add(entry.getSimilarTo());
             }
+
+            if (entry.getTypeId() != null && conceptsMergeBean.getSelectedTypeId() == null) {
+                conceptsMergeBean.setSelectedTypeId(entry.getTypeId());
+            } else if (entry.getTypeId() != null) {
+                typeWarning = true;
+            }
         }
 
+        StringBuilder errorBuilder = new StringBuilder();
+        if (posWarning) {
+            errorBuilder.append("<p>There are difference in pos within the merging concepts.</p>");
+        }
+
+        if (typeWarning) {
+            errorBuilder.append("<p>Merging concepts have different concept type.</p>");
+        }
+
+        conceptsMergeBean.setErrorMessages(errorBuilder.toString());
+
```

Comment:
Also, use an error classes instead of just one string (with field code an error code). Then you can display errors on the page nicely formatted and depending on output.